src/PerformanceTests/Utilities/Common/CompilationHelper.cs

```diff
@@ -0,0 +1,156 @@
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Composition;
+
+namespace PerformanceTests.Utilities
+{
+    public class CompilationHelper
+    {
+        public static async Task<(Project, AnalyzerOptions)> CreateProjectAsync(
+            (string, string)[] sourceFiles,
+            (string, string)[]? globalOptions,
+            string name,
+            string language,
+            string defaultPrefix,
+            string defaultExtension,
+            CompilationOptions compilationOptions,
+            ParseOptions parseOptions)
+        {
+            var projectState = ProjectState.Create(name, language, defaultPrefix, defaultExtension);
+            foreach (var (filename, content) in sourceFiles)
+            {
+                projectState.Sources.Add((defaultPrefix + filename + "." + defaultExtension, content));
+            }
+
+            var evaluatedProj = EvaluatedProjectState.Create(projectState, ReferenceAssemblies.Default);
+
+            var project = await CreateProjectAsync(evaluatedProj, compilationOptions, parseOptions);
+
+            if (globalOptions is not null)
+            {
+                var optionsProvider = new OptionsProvider(globalOptions);
+                var options = new AnalyzerOptions(ImmutableArray<AdditionalText>.Empty, optionsProvider);
+
+                return (project, options);
+
+            }
+
+            return (project, project.AnalyzerOptions);
+        }
+
+        private static async Task<Project> CreateProjectAsync(
+            EvaluatedProjectState primaryProject,
+            CompilationOptions compilationOptions,
+            ParseOptions parseOptions)
+        {
+            var projectId = ProjectId.CreateNewId(debugName: primaryProject.Name);
+            var solution = await CreateSolutionAsync(projectId, primaryProject, compilationOptions, parseOptions);
+
+            foreach (var (newFileName, source) in primaryProject.Sources)
+            {
+                var documentId = DocumentId.CreateNewId(projectId, debugName: newFileName);
+                solution = solution.AddDocument(documentId, newFileName, source, filePath: newFileName);
+            }
+
+            foreach (var (newFileName, source) in primaryProject.AdditionalFiles)
+            {
+                var documentId = DocumentId.CreateNewId(projectId, debugName: newFileName);
+                solution = solution.AddAdditionalDocument(documentId, newFileName, source, filePath: newFileName);
+            }
+
+            foreach (var (newFileName, source) in primaryProject.AnalyzerConfigFiles)
+            {
+                var documentId = DocumentId.CreateNewId(projectId, debugName: newFileName);
+                solution = solution.AddAnalyzerConfigDocument(documentId, newFileName, source, filePath: newFileName);
+            }
+
+            return solution.GetProject(projectId)!;
+        }
+
+        private static async Task<Solution> CreateSolutionAsync(
```

Comment:
💭 This is a bunch of duplication of a library we already reference. Is there a reason we can't just use the code from the test library?